cmd/clusterctl/client/cluster/mover.go

```diff
@@ -249,6 +344,70 @@ func (o *objectMover) move(graph *objectGraph, toProxy Proxy) error {
 	return setClusterPause(toProxy, clusters, false, o.dryRun)
 }
 
+func (o *objectMover) backup(graph *objectGraph, directory string) error {
+	log := logf.Log
+
+	clusters := graph.getClusters()
+	log.Info("Starting backup of Cluster API objects", "Clusters", len(clusters))
+
+	// Sets the pause field on the Cluster object in the source management cluster, so the controllers stop reconciling it.
+	log.V(1).Info("Pausing the source cluster")
+	if err := setClusterPause(o.fromProxy, clusters, true, o.dryRun); err != nil {
+		return err
+	}
+
+	// Define the move sequence by processing the ownerReference chain, so we ensure that a Kubernetes object is moved only after its owners.
+	// The sequence is bases on object graph nodes, each one representing a Kubernetes object; nodes are grouped, so bulk of nodes can be moved in parallel. e.g.
+	// - All the Clusters should be moved first (group 1, processed in parallel)
+	// - All the MachineDeployments should be moved second (group 1, processed in parallel)
+	// - then all the MachineSets, then all the Machines, etc.
+	moveSequence := getMoveSequence(graph)
+
+	// Save all objects group by group
+	log.Info("Saving files to %s", directory)
+	for groupIndex := 0; groupIndex < len(moveSequence.groups); groupIndex++ {
+		if err := o.backupGroup(moveSequence.getGroup(groupIndex), directory); err != nil {
```

Comment:
Hmmm great question. 

I'm wondering if a partial restore of objects would leave a cluster in a poor state? For example, if first a `Cluster` failed to deploy, the next group in the move sequence would be `MachineDeployments`, then `MachineSets`. Shouldn't we fail early since objects in the move sequence are being moved in a per-defined order? Since we can easily attempt a restore maneuver again and objects that already exist (aka have already been restored) are skipped, I think it's reasonable to instruct users to simply re-run restore.

If the behavior was to continue to apply objects in the move group regardless of dependent objects being there, then I think we'd get into a really weird state where owner objects do not exist and they fail to reconcile.

Reply:
that makes sense, thanks